samza-core/src/main/java/org/apache/samza/clustermanager/ContainerProcessManager.java

```diff
@@ -208,32 +210,45 @@ private static AbstractContainerAllocator buildContainerAllocator(boolean hostAf
   }
 
   public boolean shouldShutdown() {
-    log.debug("ContainerProcessManager state: Completed containers: {}, Configured containers: {}, Are there too many failed containers: {}, Is allocator thread alive: {}",
+    LOG.debug("ContainerProcessManager state: Completed containers: {}, Configured containers: {}, Are there too many failed containers: {}, Is allocator thread alive: {}",
       state.completedProcessors.get(), state.processorCount, tooManyFailedContainers ? "yes" : "no", allocatorThread.isAlive() ? "yes" : "no");
 
     if (exceptionOccurred != null) {
-      log.error("Exception in container process manager", exceptionOccurred);
+      LOG.error("Exception in container process manager", exceptionOccurred);
       throw new SamzaException(exceptionOccurred);
     }
     return tooManyFailedContainers || state.completedProcessors.get() == state.processorCount.get() || !allocatorThread.isAlive();
   }
 
   public void start() {
-    log.info("Starting the container process manager");
+    LOG.info("Starting the container process manager");
+
+    int containerRetryCount = clusterManagerConfig.getContainerRetryCount();
```

Comment:
Could this entire config handling logic be moved to a method in ClusterManagerConfig?

Reply:
got it